types/result.go

```diff
@@ -39,19 +39,27 @@ func (res Result) IsOK() bool {
 	return res.Code.IsOK()
 }
 
-// Is a version of TxResponse where the tags are StringTags rather than []byte tags
+// IndexedABCILog defines a structure containing an indexed tx ABCI message log.
+type IndexedABCILog struct {
+	MsgIndex int    `json:"msg_index"`
+	Success  bool   `json:"success"`
+	Log      string `json:"log"`
+}
+
+// TxResponse defines a structure containing relevant tx data and metadata. The
+// tags are stringified and the log is JSON decoded.
 type TxResponse struct {
-	Height    int64      `json:"height"`
-	TxHash    string     `json:"txhash"`
-	Code      uint32     `json:"code,omitempty"`
-	Data      []byte     `json:"data,omitempty"`
-	Log       string     `json:"log,omitempty"`
-	Info      string     `json:"info,omitempty"`
-	GasWanted int64      `json:"gas_wanted,omitempty"`
-	GasUsed   int64      `json:"gas_used,omitempty"`
-	Tags      StringTags `json:"tags,omitempty"`
-	Codespace string     `json:"codespace,omitempty"`
-	Tx        Tx         `json:"tx,omitempty"`
+	Height    int64            `json:"height"`
+	TxHash    string           `json:"txhash"`
+	Code      uint32           `json:"code,omitempty"`
+	Data      []byte           `json:"data,omitempty"`
+	Logs      []IndexedABCILog `json:"logs,omitempty"`
```

Comment:
Maybe we need to add a `func (ial []IndexedABCILog) String() string {}` so that these print properly in the CLI?

Reply:
Sure -- added.